### ISSUE TITLE
testing results on Semantic3D (using torch framework) do not make sense!

### ISSUE THREAD

**Issue by Asher-1** [intel-isl/Open3D-ML#203]
Initializing from scratch.
Writing summary in train_log/00001_RandLANet_Semantic3D_torch.
Started training
=== EPOCH 0/100 ===
loss train: 1.981  eval: 2.344
acc train: 0.188  eval: 0.155
iou train: 0.086  eval: 0.046
Epoch   0: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 1/100 ===
loss train: 1.747  eval: 9.761
acc train: 0.266  eval: 0.003
iou train: 0.122  eval: 0.001
=== EPOCH 2/100 ===
loss train: 1.601  eval: 10.447
acc train: 0.344  eval: 0.094
iou train: 0.162  eval: 0.006
=== EPOCH 3/100 ===
loss train: 1.432  eval: 7.275
acc train: 0.396  eval: 0.092
iou train: 0.205  eval: 0.021
=== EPOCH 4/100 ===
loss train: 1.312  eval: 1.544
acc train: 0.444  eval: 0.282
iou train: 0.240  eval: 0.138
=== EPOCH 5/100 ===
loss train: 1.282  eval: 2.148
acc train: 0.460  eval: 0.276
iou train: 0.249  eval: 0.118
Epoch   5: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 6/100 ===
loss train: 1.197  eval: 1.432
acc train: 0.484  eval: 0.290
iou train: 0.276  eval: 0.094
=== EPOCH 7/100 ===
loss train: 1.182  eval: 2.137
acc train: 0.495  eval: 0.224
iou train: 0.285  eval: 0.061
=== EPOCH 8/100 ===
loss train: 1.124  eval: 1.178
acc train: 0.511  eval: 0.422
iou train: 0.294  eval: 0.274
=== EPOCH 9/100 ===
loss train: 1.044  eval: 0.998
acc train: 0.544  eval: 0.672
iou train: 0.328  eval: 0.291
=== EPOCH 10/100 ===
loss train: 1.036  eval: 1.086
acc train: 0.562  eval: 0.500
iou train: 0.337  eval: 0.301
Epoch  10: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 11/100 ===
loss train: 1.037  eval: 1.043
acc train: 0.566  eval: 0.723
iou train: 0.338  eval: 0.304
=== EPOCH 12/100 ===
loss train: 0.956  eval: 1.005
acc train: 0.609  eval: 0.524
iou train: 0.382  eval: 0.232
=== EPOCH 13/100 ===
loss train: 0.920  eval: 1.707
acc train: 0.612  eval: 0.306
iou train: 0.380  eval: 0.142
=== EPOCH 14/100 ===
loss train: 0.943  eval: 1.862
acc train: 0.623  eval: 0.289
iou train: 0.389  eval: 0.130
=== EPOCH 15/100 ===
loss train: 0.845  eval: 1.486
acc train: 0.645  eval: 0.344
iou train: 0.417  eval: 0.161
Epoch  15: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 16/100 ===
loss train: 0.831  eval: 1.138
acc train: 0.668  eval: 0.450
iou train: 0.431  eval: 0.222
=== EPOCH 17/100 ===
loss train: 0.785  eval: 1.560
acc train: 0.675  eval: 0.390
iou train: 0.450  eval: 0.173
=== EPOCH 18/100 ===
loss train: 0.777  eval: 0.880
acc train: 0.672  eval: 0.617
iou train: 0.449  eval: 0.305
=== EPOCH 19/100 ===
loss train: 0.711  eval: 1.099
acc train: 0.709  eval: 0.529
iou train: 0.485  eval: 0.307
=== EPOCH 20/100 ===
loss train: 0.722  eval: 1.138
acc train: 0.697  eval: 0.487
iou train: 0.468  eval: 0.235
Epoch  20: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 21/100 ===
loss train: 0.708  eval: 1.007
acc train: 0.708  eval: 0.506
iou train: 0.486  eval: 0.246
=== EPOCH 22/100 ===
loss train: 0.652  eval: 0.742
acc train: 0.730  eval: 0.699
iou train: 0.515  eval: 0.291
=== EPOCH 23/100 ===
loss train: 0.682  eval: 0.905
acc train: 0.734  eval: 0.577
iou train: 0.518  eval: 0.223
=== EPOCH 24/100 ===
loss train: 0.636  eval: 0.719
acc train: 0.745  eval: 0.599
iou train: 0.531  eval: 0.327
=== EPOCH 25/100 ===
loss train: 0.593  eval: 1.294
acc train: 0.749  eval: 0.557
iou train: 0.538  eval: 0.234
Epoch  25: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 26/100 ===
loss train: 0.567  eval: 0.863
acc train: 0.757  eval: 0.571
iou train: 0.553  eval: 0.308
=== EPOCH 27/100 ===
loss train: 0.532  eval: 1.114
acc train: 0.767  eval: 0.493
iou train: 0.572  eval: 0.279
=== EPOCH 28/100 ===
loss train: 0.591  eval: 0.941
acc train: 0.770  eval: 0.651
iou train: 0.569  eval: 0.292
=== EPOCH 29/100 ===
loss train: 0.530  eval: 1.074
acc train: 0.782  eval: 0.549
iou train: 0.575  eval: 0.353
=== EPOCH 30/100 ===
loss train: 0.520  eval: 1.127
acc train: 0.783  eval: 0.601
iou train: 0.593  eval: 0.344
Epoch  30: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 31/100 ===
loss train: 0.496  eval: 1.298
acc train: 0.792  eval: 0.552
iou train: 0.605  eval: 0.246
=== EPOCH 32/100 ===
loss train: 0.452  eval: 0.995
acc train: 0.802  eval: 0.673
iou train: 0.618  eval: 0.323
=== EPOCH 33/100 ===
loss train: 0.540  eval: 0.718
acc train: 0.793  eval: 0.599
iou train: 0.589  eval: 0.291
=== EPOCH 34/100 ===
loss train: 0.465  eval: 0.728
acc train: 0.801  eval: 0.702
iou train: 0.617  eval: 0.346
=== EPOCH 35/100 ===
loss train: 0.448  eval: 0.521
acc train: 0.809  eval: 0.738
iou train: 0.620  eval: 0.411
Epoch  35: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 36/100 ===
loss train: 0.408  eval: 0.875
acc train: 0.816  eval: 0.662
iou train: 0.634  eval: 0.314
=== EPOCH 37/100 ===
loss train: 0.398  eval: 0.919
acc train: 0.808  eval: 0.634
iou train: 0.637  eval: 0.274
=== EPOCH 38/100 ===
loss train: 0.441  eval: 0.646
acc train: 0.811  eval: 0.669
iou train: 0.630  eval: 0.337
=== EPOCH 39/100 ===
loss train: 0.408  eval: 0.560
acc train: 0.832  eval: 0.762
iou train: 0.661  eval: 0.434
=== EPOCH 40/100 ===
loss train: 0.387  eval: 0.798
acc train: 0.836  eval: 0.719
iou train: 0.655  eval: 0.435
Epoch  40: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 41/100 ===
loss train: 0.441  eval: 0.609
acc train: 0.807  eval: 0.683
iou train: 0.632  eval: 0.317
=== EPOCH 42/100 ===
loss train: 0.376  eval: 0.686
acc train: 0.827  eval: 0.736
iou train: 0.652  eval: 0.439
=== EPOCH 43/100 ===
loss train: 0.386  eval: 0.477
acc train: 0.828  eval: 0.739
iou train: 0.652  eval: 0.447
=== EPOCH 44/100 ===
loss train: 0.390  eval: 0.148
acc train: 0.834  eval: 0.845
iou train: 0.658  eval: 0.622
=== EPOCH 45/100 ===
loss train: 0.361  eval: 0.523
acc train: 0.848  eval: 0.841
iou train: 0.675  eval: 0.349
Epoch  45: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 46/100 ===
loss train: 0.401  eval: 0.813
acc train: 0.829  eval: 0.657
iou train: 0.650  eval: 0.375
=== EPOCH 47/100 ===
loss train: 0.357  eval: 0.632
acc train: 0.846  eval: 0.664
iou train: 0.681  eval: 0.373
=== EPOCH 48/100 ===
loss train: 0.366  eval: 0.530
acc train: 0.830  eval: 0.790
iou train: 0.663  eval: 0.357
=== EPOCH 49/100 ===
loss train: 0.365  eval: 0.770
acc train: 0.838  eval: 0.657
iou train: 0.674  eval: 0.403
=== EPOCH 50/100 ===
loss train: 0.364  eval: 1.051
acc train: 0.835  eval: 0.578
iou train: 0.668  eval: 0.308
Epoch  50: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 51/100 ===
loss train: 0.343  eval: 0.657
acc train: 0.845  eval: 0.728
iou train: 0.681  eval: 0.460
=== EPOCH 52/100 ===
loss train: 0.327  eval: 0.952
acc train: 0.840  eval: 0.640
iou train: 0.679  eval: 0.405
=== EPOCH 53/100 ===
loss train: 0.320  eval: 0.636
acc train: 0.854  eval: 0.787
iou train: 0.691  eval: 0.444
=== EPOCH 54/100 ===
loss train: 0.360  eval: 0.548
acc train: 0.840  eval: 0.788
iou train: 0.678  eval: 0.448
=== EPOCH 55/100 ===
loss train: 0.339  eval: 0.810
acc train: 0.848  eval: 0.710
iou train: 0.687  eval: 0.388
Epoch  55: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 56/100 ===
loss train: 0.416  eval: 0.822
acc train: 0.840  eval: 0.678
iou train: 0.685  eval: 0.336
=== EPOCH 57/100 ===
loss train: 0.344  eval: 0.746
acc train: 0.842  eval: 0.704
iou train: 0.672  eval: 0.408
=== EPOCH 58/100 ===
loss train: 0.350  eval: 0.377
acc train: 0.850  eval: 0.774
iou train: 0.687  eval: 0.444
=== EPOCH 59/100 ===
loss train: 0.315  eval: 0.775
acc train: 0.851  eval: 0.653
iou train: 0.682  eval: 0.484
=== EPOCH 60/100 ===
loss train: 0.348  eval: 0.400
acc train: 0.837  eval: 0.731
iou train: 0.681  eval: 0.389
Epoch  60: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 61/100 ===
loss train: 0.297  eval: 0.453
acc train: 0.859  eval: 0.778
iou train: 0.700  eval: 0.492
=== EPOCH 62/100 ===
loss train: 0.350  eval: 0.909
acc train: 0.840  eval: 0.610
iou train: 0.679  eval: 0.292
=== EPOCH 63/100 ===
loss train: 0.342  eval: 0.608
acc train: 0.856  eval: 0.738
iou train: 0.696  eval: 0.471
=== EPOCH 64/100 ===
loss train: 0.355  eval: 0.548
acc train: 0.847  eval: 0.793
iou train: 0.682  eval: 0.401
=== EPOCH 65/100 ===
loss train: 0.316  eval: 0.703
acc train: 0.852  eval: 0.715
iou train: 0.684  eval: 0.406
Epoch  65: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 66/100 ===
loss train: 0.329  eval: 0.578
acc train: 0.853  eval: 0.710
iou train: 0.694  eval: 0.443
=== EPOCH 67/100 ===
loss train: 0.290  eval: 0.536
acc train: 0.866  eval: 0.714
iou train: 0.713  eval: 0.385
=== EPOCH 68/100 ===
loss train: 0.292  eval: 0.675
acc train: 0.859  eval: 0.668
iou train: 0.710  eval: 0.410
=== EPOCH 69/100 ===
loss train: 0.287  eval: 0.480
acc train: 0.858  eval: 0.773
iou train: 0.712  eval: 0.458
=== EPOCH 70/100 ===
loss train: 0.291  eval: 0.793
acc train: 0.865  eval: 0.739
iou train: 0.713  eval: 0.495
Epoch  70: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 71/100 ===
loss train: 0.290  eval: 0.868
acc train: 0.855  eval: 0.643
iou train: 0.697  eval: 0.432
=== EPOCH 72/100 ===
loss train: 0.320  eval: 0.453
acc train: 0.857  eval: 0.798
iou train: 0.708  eval: 0.475
=== EPOCH 73/100 ===
loss train: 0.340  eval: 0.723
acc train: 0.840  eval: 0.559
iou train: 0.687  eval: 0.345
=== EPOCH 74/100 ===
loss train: 0.300  eval: 0.796
acc train: 0.851  eval: 0.618
iou train: 0.702  eval: 0.355
=== EPOCH 75/100 ===
loss train: 0.284  eval: 0.541
acc train: 0.867  eval: 0.766
iou train: 0.715  eval: 0.374
Epoch  75: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 76/100 ===
loss train: 0.326  eval: 0.495
acc train: 0.859  eval: 0.807
iou train: 0.700  eval: 0.387
=== EPOCH 77/100 ===
loss train: 0.327  eval: 0.573
acc train: 0.861  eval: 0.733
iou train: 0.708  eval: 0.403
=== EPOCH 78/100 ===
loss train: 0.340  eval: 0.833
acc train: 0.864  eval: 0.719
iou train: 0.708  eval: 0.465
=== EPOCH 79/100 ===
loss train: 0.280  eval: 0.711
acc train: 0.864  eval: 0.635
iou train: 0.706  eval: 0.423
=== EPOCH 80/100 ===
loss train: 0.331  eval: 0.520
acc train: 0.864  eval: 0.749
iou train: 0.702  eval: 0.441
Epoch  80: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 81/100 ===
loss train: 0.304  eval: 0.340
acc train: 0.864  eval: 0.816
iou train: 0.705  eval: 0.542
=== EPOCH 82/100 ===
loss train: 0.277  eval: 0.974
acc train: 0.866  eval: 0.613
iou train: 0.710  eval: 0.416
=== EPOCH 83/100 ===
loss train: 0.313  eval: 0.591
acc train: 0.854  eval: 0.767
iou train: 0.691  eval: 0.460
=== EPOCH 84/100 ===
loss train: 0.334  eval: 0.473
acc train: 0.859  eval: 0.801
iou train: 0.696  eval: 0.489
=== EPOCH 85/100 ===
loss train: 0.325  eval: 0.295
acc train: 0.858  eval: 0.839
iou train: 0.704  eval: 0.524
Epoch  85: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 86/100 ===
loss train: 0.291  eval: 1.024
acc train: 0.859  eval: 0.545
iou train: 0.700  eval: 0.342
=== EPOCH 87/100 ===
loss train: 0.305  eval: 0.718
acc train: 0.862  eval: 0.674
iou train: 0.706  eval: 0.419
=== EPOCH 88/100 ===
loss train: 0.289  eval: 0.502
acc train: 0.860  eval: 0.750
iou train: 0.708  eval: 0.472
=== EPOCH 89/100 ===
loss train: 0.310  eval: 0.467
acc train: 0.857  eval: 0.730
iou train: 0.695  eval: 0.510
=== EPOCH 90/100 ===
loss train: 0.327  eval: 0.510
acc train: 0.854  eval: 0.689
iou train: 0.699  eval: 0.437
Epoch  90: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 91/100 ===
loss train: 0.293  eval: 0.363
acc train: 0.864  eval: 0.754
iou train: 0.714  eval: 0.467
=== EPOCH 92/100 ===
loss train: 0.293  eval: 0.439
acc train: 0.871  eval: 0.780
iou train: 0.710  eval: 0.504
=== EPOCH 93/100 ===
loss train: 0.280  eval: 0.452
acc train: 0.862  eval: 0.834
iou train: 0.705  eval: 0.536
=== EPOCH 94/100 ===
loss train: 0.312  eval: 0.657
acc train: 0.865  eval: 0.732
iou train: 0.706  eval: 0.501
=== EPOCH 95/100 ===
loss train: 0.276  eval: 0.237
acc train: 0.871  eval: 0.855
iou train: 0.715  eval: 0.600
Epoch  95: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
=== EPOCH 96/100 ===
loss train: 0.291  eval: 0.739
acc train: 0.860  eval: 0.562
iou train: 0.701  eval: 0.295
=== EPOCH 97/100 ===
loss train: 0.328  eval: 0.538
acc train: 0.857  eval: 0.699
iou train: 0.708  eval: 0.463
=== EPOCH 98/100 ===
loss train: 0.271  eval: 0.784
acc train: 0.861  eval: 0.704
iou train: 0.713  eval: 0.404
=== EPOCH 99/100 ===
loss train: 0.299  eval: 0.249
acc train: 0.854  eval: 0.900
iou train: 0.693  eval: 0.363
=== EPOCH 100/100 ===
loss train: 0.282  eval: 0.577
acc train: 0.850  eval: 0.728
iou train: 0.698  eval: 0.359
Epoch 100: save ckpt to ./logs/RandLANet_Semantic3D_torch/checkpoint
ckpt_path not given. Restore from the latest ckpt
Loading checkpoint ./logs/RandLANet_Semantic3D_torch/checkpoint/ckpt_00100.pth

Tesing test data on Semantic3D dataset by using examples/visualize.py script，but the visualization result is as follows:
![微信图片编辑_20201230100928](https://user-images.githubusercontent.com/16451440/103325524-219e6080-4a87-11eb-9536-ba8a9a062062.jpg)

**Comment by YilingQiao** [comment]
Hi, Amar! The training script for this dataset still has some issues and we are working in progress.
We will let you know when the training is fixed. Could you please try the pretrained weights?

**Comment by Asher-1** [comment]
thanks
